string new_string
---
string output_string
